src/queries/catalog/call-context-query/call-context-query-executor.ts

```diff
@@ -203,6 +203,22 @@ export function executeCallContextQueries({ graph, ast }: BasicQueryData, querie
 		}
 
 		for(const query of promotedQueries.filter(q => q.callName.test(info.name))) {
+			const file = ast.idMap.get(nodeId)?.info.file;
+			filterCheck: if(query.fileFilter !== undefined) {
+				if(file === undefined) {
+					if(query.includeUndefinedFiles ?? true) {
+						break filterCheck; // although the node's file is unknown, we continue with the query
+					} else {
+						continue; // the file is unknown, and we don't want to include it; skip to the next query
+					}
+				}
+				if(new RegExp(query.fileFilter).test(file)) {
```

Comment:
The `RegExp(query.fileFilter)` should happen in the promote query stage to avoid the repeated re-creation